# Final Project 
## Ojus Sindwani

## Aim of the Project 
Undertake a financial/accounting analysis of a large set of U.S. companies during 5 key stock market crashes over the past few decades and explain determinants of stock returns.

1. **1987 Stock Market Crash** (Sept 1987 – Dec 1987)
2. **Dot Com Bubble and Crash** ( Jan 2000 – Oct 2002)
3. **Great Recession & Financial Crisis** (Jan 2008 – Feb 2009) 
4. **The Covid Shock** (Feb 2020 – March 2020)
5. **“The End of Easy Money”**(Jan 2022 – Sep 2022)



In [95]:
#Importing the necessary libraries and packages 
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm

In [96]:
#mounting the drive 
from google.colab import drive
drive.mount('/gdrive')

Drive already mounted at /gdrive; to attempt to forcibly remount, call drive.mount("/gdrive", force_remount=True).


### Uploading the G Sector Files 

In [97]:
gsector_1987 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/GSECTOR-1986.csv')
gsector_2001 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/GSECTOR-2000.csv')
gsector_2008 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/GSECTOR-2007.csv')
gsector_2020 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/GSECTOR-2019.csv')
gsector_2022 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/GSECTOR-2021.csv')

An explanation of the G-Sectors is given as below:

**`10.0`**
Energy

**`15.0`**
Materials

**`20.0`**
Industrials

**`25.0`**
Consumer Discretionary

**`30.0`**
Consumer Staples

**`35.0`**
Health Care

**`40.0`**
Financials

**`45.0`**
Information Technology

**`50.0`**
Communication Services

**`55.0`**
Utilities

**`60.0`**
Real Estate


In [156]:
gsector_1987.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7219 entries, 0 to 7218
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   gvkey    7219 non-null   int64  
 1   fyear    7219 non-null   int64  
 2   TICKER   7219 non-null   object 
 3   gsector  7210 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 225.7+ KB


### Uploading files containing financial ratios

These files have been derived from assigment 3

In [99]:
ratios_1987 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/financial_ratios_1987.csv')
ratios_2001 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/financial_ratios_2001.csv')
ratios_2008 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/financial_ratios_2008.csv')
ratios_2020 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/financial_ratios_2020.csv')
ratios_2022 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/financial_ratios_2022.csv')

These files contain various financial ratios for each of the stock market crashes and the average returns for those crashes. The following analysis has already been performed on these datasets in Assignment 3 for the purpose of calculating regression.


1.   **`Merging Data`** Merging financial data with average returns for the months leading up to the 5 stock market crashes.
2.   **`Treating Missing Values`** Running logistic regression for Current Assets, Current Liabilites, Total Liabilities, Long-Term Debt, Cost of Goods Sold, and Retained Earnings. 
3.   **`Calculating financial ratios`** Calculating financial ratios like Net Profit Margin, Gross Profit Margin, Liquidity ratio, Leverage, Debt/Total Assets, Asset Turnover Ratio, and ROE. 
4.   **`Regression`** Regressing various financial ratios on avergae returns and interpreting the results.




In [157]:
ratios_1987.head()

gvkey    datadate  fyear indfmt consol popsrc datafmt TICKER curcd  \
0   1003  1987-01-31   1986   INDL      C      D     STD   ANTQ   USD   
1   1004  1987-05-31   1986   INDL      C      D     STD    AIR   USD   
2   1011  1986-12-31   1986   INDL      C      D     STD   ACSE   USD   
3   1016  1986-10-31   1986   INDL      C      D     STD   AECE   USD   
4   1017  1987-02-28   1986   INDL      C      D     STD  AELNA   USD   

       act  ...       RET  Net Profit Margin  Gross Profit Margin  \
0   13.090  ... -0.126467           0.021841             0.473422   
1  168.950  ... -0.058069           0.051514             0.247015   
2    2.385  ... -0.069427          -0.145579            -0.006851   
3   18.715  ...  0.014061           0.015301             0.360856   
4   52.080  ... -0.099851           0.030642             0.263189   

   Total Debt/Equity  Return on Equity  Liquidity       ROA  \
0           0.955752          0.106329   3.875074  0.054367   
1           0.750361          0.114370   2.078004  0.065341   
2           0.620234         -0.169449   2.021186 -0.104583   
3           1.231830          0.080062   2.071389  0.035873   
4           0.624524          0.059649   2.012365  0.036718   

   Debt/Total Assets  Asset Turnover Ratio    at_log  
0           0.488688              2.489236  2.680062  
1           0.428689              1.268411  5.459973  
2           0.382805              0.718394  1.872110  
3           0.551937              2.344443  3.314695  
4           0.384435              1.198293  4.504764  

[5 rows x 30 columns]

### Uploading files containing the FF output for 5 years 

These files have been derived from assignment 2

In [100]:
ff_1987 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/FF-Output-1983-1987.csv')
ff_2001 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/FF-Output-1997-2001.csv')
ff_2008 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/FF-Output-2004-2008.csv')
ff_2020 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/FF-Output-2016-2020.csv')
ff_2022 = pd.read_csv(r'/gdrive/MyDrive/Finance Final Project/FF-Output-2018-2022.csv')

In [158]:
ff_1987.head()

Unnamed: 0 TICKER     const     mktrf       smb       hml
0           1    CNK  0.008628  0.881037  0.581720 -0.000531
1           2   GENE  0.038865  1.157391  1.382452 -1.254665
2           3   GAMA -0.005431  0.772968 -0.073854  0.189572
3           4  TCOMA  0.029071  0.926688  0.038847 -1.166874
4           5    WOC -0.011008  0.652339  0.582038  0.305336

# 1987 Stock Market Crash 

The stock market crash of 1987, also known as Black Monday, occurred on October 19th, 1987. The crash was a sudden and severe drop in stock prices, with the Dow Jones Industrial Average falling by over 22% in one day. The crash was caused by a combination of factors, including overvalued stocks, a decline in international markets, and computerized trading programs. 

## Merging the three files together 

In [101]:
# Renaming tic to TICKER in the gsector file
gsector_1987 = gsector_1987.rename(columns={'tic': 'TICKER'})

In [102]:
# Merge the DataFrames based on the 'TICKER' column
merged_df = pd.merge(gsector_1987, ratios_1987, on='TICKER')
merged_df = pd.merge(merged_df, ff_1987, on='TICKER')

In [103]:
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 880 entries, 0 to 879
Data columns (total 38 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gvkey_x               880 non-null    int64  
 1   fyear_x               880 non-null    int64  
 2   TICKER                880 non-null    object 
 3   gsector               880 non-null    float64
 4   gvkey_y               880 non-null    int64  
 5   datadate              880 non-null    object 
 6   fyear_y               880 non-null    int64  
 7   indfmt                880 non-null    object 
 8   consol                880 non-null    object 
 9   popsrc                880 non-null    object 
 10  datafmt               880 non-null    object 
 11  curcd                 880 non-null    object 
 12  act                   880 non-null    float64
 13  at                    880 non-null    float64
 14  che                   880 non-null    float64
 15  cogs                  8

In [104]:
# Getting a sense of industry indicators 
sector_counts = merged_df.groupby('gsector').size()
sector_counts

gsector
10.0     52
15.0     66
20.0    211
25.0    147
30.0     60
35.0     67
40.0     77
45.0    117
50.0     14
55.0     56
60.0     13
dtype: int64

## Run OLS regression with the following three main categories 


1.   **Risk Exposures:** This included systematic risk (market risk), SML (Small Minus Big), which is the size risk, and HML (High minus Low), which is the Value vs Growth risk. 

2.   **Financial Ratios:** This includes various financial ratios, including but not limited to Liquidity, Leverage, Asset Turnover Ratio, Net and Gross Profit Margin, ROE and ROA. 

3.   **Industry Indicators:** This indicates which industries had the highest amount of affect on stock returns. 



### 1987- Regression 1

Regression using all 3 Categories 

In [105]:
# General regression code:
X1 = merged_df[['Gross Profit Margin', 'Debt/Total Assets', 'Asset Turnover Ratio', 'at_log']]
X2 = pd.get_dummies(merged_df['gsector'], prefix='sector')
X3 = merged_df[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2, X3], axis=1)

y = merged_df['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.242
Model:                            OLS   Adj. R-squared:                  0.227
Method:                 Least Squares   F-statistic:                     16.20
Date:                Sat, 06 May 2023   Prob (F-statistic):           1.39e-41
Time:                        12:27:41   Log-Likelihood:                 1397.9
No. Observations:                 880   AIC:                            -2760.
Df Residuals:                     862   BIC:                            -2674.
Df Model:                          17                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   -0.0087 

The regression model shows the relationship between the dependent variable, RET(stock returns), and various independent variables. The R-squared value suggests that 24.2% of the variance in the dependent variable is explained by the model, with an adjusted R-Squared. The coefficients of the independent variables show their impact on the dependent variable, with Gross Profit Margin, Debt/Total Assets, Asset Turnover Ratio, sector_10.0, sector_15.0, sector_25.0, sector_45.0, mktrf, and smb having statistically significant impacts. The model has some potential issues with multicollinearity and singularity, therefore, further regressions are performed to reduce such issues.

### 1987- Regression 2

Regression using only financial ratios and industry indicators 

In [106]:
# General regression code:
X1 = merged_df[['Gross Profit Margin', 'Debt/Total Assets', 'Asset Turnover Ratio', 'at_log']]
X2 = pd.get_dummies(merged_df['gsector'], prefix='sector')

X = pd.concat([X1, X2], axis=1)

y = merged_df['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.109
Model:                            OLS   Adj. R-squared:                  0.095
Method:                 Least Squares   F-statistic:                     7.574
Date:                Sat, 06 May 2023   Prob (F-statistic):           3.46e-15
Time:                        12:27:41   Log-Likelihood:                 1326.7
No. Observations:                 880   AIC:                            -2623.
Df Residuals:                     865   BIC:                            -2552.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   -0.0507 

The OLS regression model suggests that Gross Profit Margin, Debt/Total Assets, Asset Turnover Ratio, and some industry sectors have significant effects on the stock returns of the companies in the sample. However, the overall explanatory power of the model is weak, as indicated by the low R-squared value of 10.9%, and an adjusted R-Squared value 9.5%. Additionally, there may be issues of multicollinearity, and the design matrix may be singular, which could affect the reliability of the coefficient estimates. As risk exposures were excluded, the R2 dropped significantly indicating that 13.3% variance in the model is explained by risk exposures.

### 1987- Regression 3

Regression using only financial ratios and risk exposures 

In [107]:
# General regression code:
X1 = merged_df[['Gross Profit Margin', 'Debt/Total Assets', 'Asset Turnover Ratio', 'at_log']]
X2 = merged_df[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.213
Model:                            OLS   Adj. R-squared:                  0.207
Method:                 Least Squares   F-statistic:                     33.81
Date:                Sat, 06 May 2023   Prob (F-statistic):           8.75e-42
Time:                        12:27:41   Log-Likelihood:                 1381.5
No. Observations:                 880   AIC:                            -2747.
Df Residuals:                     872   BIC:                            -2709.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   -0.0114 

This is a multiple linear regression with seven independent variables. The dependent variable is the RET (stock returns) and the independent variables are Gross Profit Margin, Debt/Total Assets, Asset Turnover Ratio, at_log, mktrf, smb, and hml. The R-squared value is 0.213, indicating that the model explains 21.3% of the variance in the dependent variable, with an adjusted R-Squareded of 20.7%. The t-tests and p-values indicate the statistical significance of the coefficients. The regression suggests that Gross Profit Margin, Debt/Total Assets, Asset Turnover Ratio, at_log, mktrf, and smb have significant impact on the return, while hml is borderline significant. This regression dropped the financial ratios, indicating that 2.9% of the variance was explained by the financial ratios.

### 1987- Regression 4

Regression using only risk exposure

In [108]:
#General regression code:
X = merged_df[['mktrf', 'smb', 'hml']]
y = merged_df['RET']
# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

#list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.191
Model:                            OLS   Adj. R-squared:                  0.188
Method:                 Least Squares   F-statistic:                     68.74
Date:                Sat, 06 May 2023   Prob (F-statistic):           6.36e-40
Time:                        12:27:41   Log-Likelihood:                 1368.9
No. Observations:                 880   AIC:                            -2730.
Df Residuals:                     876   BIC:                            -2711.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.0006      0.005     -0.118      0.9

This is a regression model with three independent variables, namely mktrf, smb, and hml for the risk exposures, and a dependent variable RET (stock returns). The model has an R-squared value of 0.191, indicating that the independent variables explain approximately 19.1% of the variance in the dependent variable, with an adjuted R-Squared of 18.8%. The coefficients of the independent variables suggest that mktrf and smb have negative relationships with RET, while hml has a positive relationship. Overall, the model can be used to predict the value of RET based on the values of the independent variables.

### 1987- Regression 5

Regression using only industry indicators

In [138]:
X_all_sectors = pd.get_dummies(merged_df['gsector'], prefix='sector')
y = merged_df['RET']
# Perform the regression using statsmodels
model_all_sectors = sm.OLS(y, X_all_sectors).fit()

# Identify the industry with the coefficient value closest to zero
reference_sector = model_all_sectors.params.abs().idxmin()

print(reference_sector)

sector_60.0


In [139]:
# Create dummy variables for the industry sectors
X_all_sectors = pd.get_dummies(merged_df['gsector'], prefix='sector')
y = merged_df['RET']

# Exclude the reference sector's dummy variable from the independent variables
X = X_all_sectors.drop(columns=[reference_sector])

# Add a constant to the independent variables
X = sm.add_constant(X)

# Perform the regression using statsmodels without the reference sector's dummy variable
model_5 = sm.OLS(y, X).fit()

# Print the regression output
print(model_5.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.068
Model:                            OLS   Adj. R-squared:                  0.057
Method:                 Least Squares   F-statistic:                     6.311
Date:                Sat, 06 May 2023   Prob (F-statistic):           2.15e-09
Time:                        12:30:28   Log-Likelihood:                 1306.7
No. Observations:                 880   AIC:                            -2591.
Df Residuals:                     869   BIC:                            -2539.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0346      0.015     -2.259      

The regression model examines the relationship between the stock returns (dependent variable) and industry sectors (independent variables) using Ordinary Least Squares (OLS) method.

The initial model includes dummy variables for all industry sectors, and the industry with the coefficient value closest to zero is identified as the reference sector. Then, a new model is created by excluding the reference sector's dummy variable from the independent variables.

The final model shows that the adjusted R-squared value is 0.057, indicating that the independent variables can explain 5.7% of the variation in the dependent variable.

Among the remaining industry sectors, the coefficients for sectors 10, 25, 40, and 45 are statistically significant at the 5% level, indicating that these sectors have a significant impact on stock returns. The coefficient for sector 15 is not statistically significant. The negative coefficients for all significant sectors suggest that stocks in these sectors have a negative impact on returns. The intercept coefficient is also negative and statistically significant, indicating that the expected return is negative even when all the independent variables are zero.

### 1987- Conclusion

`Thus we note that the maximum variability in the data is explained by the risk exposure`

This too in this case, by the systematic risk and the size risk(Small minus Big).

# Dot Com Bubble and Crash

The dot com bubble crash of 2001 was a significant market downturn in the technology sector, fueled by the excessive speculation and overvaluation of internet-related stocks. This bubble began to burst in March 2000, leading to a sharp decline in stock prices of companies in the technology industry. By 2001, many dot com companies went bankrupt, resulting in massive job losses and a decline in overall economic growth. The crash was a painful reminder of the risks associated with speculative bubbles and market irrationality.

### Merging the three files together 

In [109]:
# Renaming tic to TICKER in the gsector file
gsector_2001 = gsector_2001.rename(columns={'tic': 'TICKER'})

In [110]:
# Merge the DataFrames based on the 'TICKER' column
merged_df_2001 = pd.merge(gsector_2001, ratios_2001, on='TICKER')
merged_df_2001 = pd.merge(merged_df_2001, ff_2001, on='TICKER')

In [111]:
merged_df_2001.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1973 entries, 0 to 1972
Data columns (total 38 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gvkey_x               1973 non-null   int64  
 1   fyear_x               1973 non-null   int64  
 2   TICKER                1973 non-null   object 
 3   gsector               1972 non-null   float64
 4   gvkey_y               1973 non-null   int64  
 5   datadate              1973 non-null   object 
 6   fyear_y               1973 non-null   int64  
 7   indfmt                1973 non-null   object 
 8   consol                1973 non-null   object 
 9   popsrc                1973 non-null   object 
 10  datafmt               1973 non-null   object 
 11  curcd                 1973 non-null   object 
 12  act                   1973 non-null   float64
 13  at                    1973 non-null   float64
 14  che                   1973 non-null   float64
 15  cogs                 

## Run OLS regression with the following three main categories 


1.   **Risk Exposures:** This included systematic risk (market risk), SML (Small Minus Big), which is the size risk, and HML (High minus Low), which is the Value vs Growth risk. 

2.   **Financial Ratios:** This includes various financial ratios, including but not limited to Liquidity, Leverage, Asset Turnover Ratio, Net and Gross Profit Margin, ROE and ROA. 

3.   **Industry Indicators:** This indicates which industries had the highest amount of affect on stock returns. 



### 2001- Regression 1 

Regression using industry indicators, financial ratios and risk exposures.

In [112]:
# General regression code:
X1 = merged_df_2001[['Net Profit Margin', 'Gross Profit Margin', 'ROA', 'Debt/Total Assets', 'Asset Turnover Ratio']]
X2 = pd.get_dummies(merged_df_2001['gsector'], prefix='sector')
X3 = merged_df_2001[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2, X3], axis=1)

y = merged_df_2001['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.352
Model:                            OLS   Adj. R-squared:                  0.346
Method:                 Least Squares   F-statistic:                     55.87
Date:                Sat, 06 May 2023   Prob (F-statistic):          2.62e-168
Time:                        12:27:41   Log-Likelihood:                 3203.7
No. Observations:                1973   AIC:                            -6367.
Df Residuals:                    1953   BIC:                            -6256.
Df Model:                          19                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                    0.0180 

The regression model examines the relationship between stock returns (RET) and various independent variables. The R-squared value of 0.352 indicates that 35.2% of the variance in the dependent variable is explained by the model, with an adjusted R-squared of 0.346. The coefficients of the independent variables indicate that Gross Profit Margin, Debt/Total Assets, Asset Turnover Ratio, sector_10.0, sector_15.0, sector_25.0, sector_45.0, mktrf, and hml have a statistically significant impact on the dependent variable. However, the model may have some issues with multicollinearity and singularity, and further regressions are required to address these issues.

### 2001- Regression 2

Regression using only industry indicators and risk exposure 

In [113]:
# General regression code:
X1 = pd.get_dummies(merged_df_2001['gsector'], prefix='sector')
X2 = merged_df_2001[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df_2001['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.348
Model:                            OLS   Adj. R-squared:                  0.343
Method:                 Least Squares   F-statistic:                     74.58
Date:                Sat, 06 May 2023   Prob (F-statistic):          4.29e-170
Time:                        12:27:41   Log-Likelihood:                 3197.1
No. Observations:                1973   AIC:                            -6364.
Df Residuals:                    1958   BIC:                            -6280.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const           0.0275      0.048      0.572      

This is an OLS regression summary, showing the results of a regression model that estimates the relationship between the dependent variable "RET" and the independent variables "const", "sector_10.0", "sector_15.0", "sector_20.0", "sector_25.0", "sector_30.0", "sector_35.0", "sector_40.0", "sector_45.0", "sector_50.0", "sector_55.0", "sector_60.0", "mktrf", "smb", and "hml". 

The R-squared value of 0.348 indicates that 34.8% of the variance in the dependent variable is explained by the independent variables. The adjusted R-squared value of 0.343 is slightly lower, indicating that the model has not overfit the data.

Removing financial ratios from the regression reduced 0.4% explanatory power from the regression.

### 2001- Regression 3 

Regression using only risk exposure and financial ratios

In [114]:
# General regression code:
X1 = merged_df_2001[['Net Profit Margin', 'Gross Profit Margin', 'ROA', 'Debt/Total Assets', 'Asset Turnover Ratio']]
X2 = merged_df_2001[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df_2001['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.340
Model:                            OLS   Adj. R-squared:                  0.338
Method:                 Least Squares   F-statistic:                     126.7
Date:                Sat, 06 May 2023   Prob (F-statistic):          2.21e-171
Time:                        12:27:41   Log-Likelihood:                 3186.0
No. Observations:                1973   AIC:                            -6354.
Df Residuals:                    1964   BIC:                            -6304.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                    0.0041 

This is an OLS regression summary, showing the results of a regression model that estimates the relationship between the dependent variable "RET" and the independent variables "const", "Net Profit Margin", "Gross Profit Margin", "ROA", "Debt/Total Assets", "Asset Turnover Ratio", "mktrf", "smb", and "hml".

The R-squared value of 0.340 indicates that 34.0% of the variance in the dependent variable is explained by the independent variables. The adjusted R-squared value of 0.338 is slightly lower, indicating that the model has not overfit the data.

Removing gsector indicators reduced the explanatory power by only 0.8%.

### 2001- Regression 4 

Regression using only risk exposure

In [115]:
#General regression code:
X = merged_df_2001[['mktrf', 'smb', 'hml']]
y = merged_df_2001['RET']
# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

#list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.337
Model:                            OLS   Adj. R-squared:                  0.336
Method:                 Least Squares   F-statistic:                     333.4
Date:                Sat, 06 May 2023   Prob (F-statistic):          4.81e-175
Time:                        12:27:41   Log-Likelihood:                 3180.7
No. Observations:                1973   AIC:                            -6353.
Df Residuals:                    1969   BIC:                            -6331.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0134      0.002      6.440      0.0

This is a multiple linear regression model with three independent variables: mktrf, smb, and hml. The dependent variable is RET. The model has a significant overall fit with an R-squared of 0.337, meaning 33.7% of the variance in RET is explained by the three independent variables. The coefficients for mktrf and hml are significant at the 0.05 level, while the coefficient for smb is not significant at the 0.05 level. The model's intercept is also significant at the 0.05 level. The model assumes that the covariance matrix of the errors is correctly specified.



### 2001- Regression 5

This regression only takes into account the industry indicators.

In [140]:
X_all_sectors = pd.get_dummies(merged_df_2001['gsector'], prefix='sector')
y = merged_df_2001['RET']
# Perform the regression using statsmodels
model_all_sectors = sm.OLS(y, X_all_sectors).fit()

# Identify the industry with the coefficient value closest to zero
reference_sector = model_all_sectors.params.abs().idxmin()

print(reference_sector)

sector_35.0


In [141]:
# Create dummy variables for the industry sectors
X_all_sectors = pd.get_dummies(merged_df_2001['gsector'], prefix='sector')
y = merged_df_2001['RET']

# Exclude the reference sector's dummy variable from the independent variables
X = X_all_sectors.drop(columns=[reference_sector])

# Add a constant to the independent variables
X = sm.add_constant(X)

# Perform the regression using statsmodels without the reference sector's dummy variable
model_5 = sm.OLS(y, X).fit()

# Print the regression output
print(model_5.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.148
Model:                            OLS   Adj. R-squared:                  0.144
Method:                 Least Squares   F-statistic:                     34.11
Date:                Sat, 06 May 2023   Prob (F-statistic):           9.67e-62
Time:                        12:35:50   Log-Likelihood:                 2933.6
No. Observations:                1973   AIC:                            -5845.
Df Residuals:                    1962   BIC:                            -5784.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const           0.0004      0.004      0.115      

The regression output shows the results of a linear regression model that aims to predict the return (RET) of companies based on their industry sectors. The model uses dummy variables to represent the different industry sectors, and it excludes one reference sector to avoid multicollinearity.

The R-squared value of the model is 0.148, indicating that only 14.8% of the variation in the dependent variable is explained by the independent variables. The adjusted R-squared value is slightly lower, which suggests that the model might be overfitting the data.

Some of the coefficients have p-values higher than 0.05, indicating that they are not statistically significant. In particular, the coefficient for sector_50.0 has a p-value of 0.190, which suggests that it is not a good predictor of the dependent variable.

The coefficient for the reference sector (sector_20.0) is not shown in the output, as it was excluded from the model. The constant term is also included in the model, with a coefficient of 0.0004. Overall, the model suggests that the industry sectors of companies have a significant but relatively weak relationship with their returns.





### 2001- Conclusion 

**`Therefore, the risk exposures explain the most variance in the stock returns with an explnantory power of 33.7%`**

# Great Recession and Financial Crisis

The Great Recession of 2008 was a severe economic downturn that lasted from late 2007 to mid-2009, triggered by a collapse in the housing market and the subprime mortgage industry in the United States. It caused widespread job losses, foreclosures, and a significant decline in economic activity across the globe, leading to a financial crisis that affected the banking sector and financial markets worldwide. The recession had a lasting impact on many economies and societies, and policymakers implemented a range of measures to prevent another such event from occurring.

### Merging the three files together 

## Run OLS regression with the following three main categories 


1.   **Risk Exposures:** This included systematic risk (market risk), SML (Small Minus Big), which is the size risk, and HML (High minus Low), which is the Value vs Growth risk. 

2.   **Financial Ratios:** This includes various financial ratios, including but not limited to Liquidity, Leverage, Asset Turnover Ratio, Net and Gross Profit Margin, ROE and ROA. 

3.   **Industry Indicators:** This indicates which industries had the highest amount of affect on stock returns. 



In [116]:
# Renaming tic to TICKER in the gsector file
gsector_2008 = gsector_2008.rename(columns={'tic': 'TICKER'})

In [117]:
# Merge the DataFrames based on the 'TICKER' column
merged_df_2008 = pd.merge(gsector_2008, ratios_2008, on='TICKER')
merged_df_2008 = pd.merge(merged_df_2008, ff_2008, on='TICKER')

In [118]:
merged_df_2008.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2440 entries, 0 to 2439
Data columns (total 38 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gvkey_x               2440 non-null   int64  
 1   fyear_x               2440 non-null   int64  
 2   TICKER                2440 non-null   object 
 3   gsector               2440 non-null   float64
 4   gvkey_y               2440 non-null   int64  
 5   datadate              2440 non-null   object 
 6   fyear_y               2440 non-null   int64  
 7   indfmt                2440 non-null   object 
 8   consol                2440 non-null   object 
 9   popsrc                2440 non-null   object 
 10  datafmt               2440 non-null   object 
 11  curcd                 2440 non-null   object 
 12  act                   2440 non-null   float64
 13  at                    2440 non-null   float64
 14  che                   2440 non-null   float64
 15  cogs                 

#### 2008- Regression 1

Regression using financial ratios, risk exposures and industry indicators

In [119]:
# General regression code:
X1 = merged_df_2008[['Total Debt/Equity', 'ROA',  'Asset Turnover Ratio']]
X2 = pd.get_dummies(merged_df_2008['gsector'], prefix='sector')
X3 = merged_df_2008[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2, X3], axis=1)

y = merged_df_2008['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.698
Model:                            OLS   Adj. R-squared:                  0.696
Method:                 Least Squares   F-statistic:                     350.8
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:42   Log-Likelihood:                 3446.3
No. Observations:                2440   AIC:                            -6859.
Df Residuals:                    2423   BIC:                            -6760.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                    0.0156 

The OLS regression model examines the relationship between stock returns (RET) and various independent variables, including financial ratios and sector dummies. The model's R-squared value indicates that about 69.8% of the variation in the dependent variable is explained by the model, with an adjusted R-squared of 69.6%. The coefficients of the independent variables show that only ROA, sector_10.0, sector_40.0, sector_55.0, mktrf, smb, and hml have statistically significant impacts on stock returns. The model may suffer from multicollinearity and singularity issues, indicating a need for further analysis as done below.

#### 2008- Regression 2

Regression using financial ratios and industry indicators

In [120]:
# General regression code:
X1 = merged_df_2008[['Total Debt/Equity', 'ROA',  'Asset Turnover Ratio']]
X2 = pd.get_dummies(merged_df_2008['gsector'], prefix='sector')

X = pd.concat([X1, X2], axis=1)

y = merged_df_2008['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.130
Model:                            OLS   Adj. R-squared:                  0.126
Method:                 Least Squares   F-statistic:                     27.95
Date:                Sat, 06 May 2023   Prob (F-statistic):           1.38e-64
Time:                        12:27:42   Log-Likelihood:                 2153.9
No. Observations:                2440   AIC:                            -4280.
Df Residuals:                    2426   BIC:                            -4199.
Df Model:                          13                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   -0.1292 

This is the output of a multiple linear regression analysis where the dependent variable is RET (return) and 13 independent variables have been used in the model. 

The adjusted R-squared value for the model is 0.126, indicating that approximately 13% of the variation in the dependent variable can be explained by the independent variables included in the model. The adjusted R-squared value takes into account the number of independent variables in the model and adjusts the R-squared value accordingly. In this case, the adjusted R-squared value is slightly lower than the R-squared value of 0.130, which is the proportion of the variation in the dependent variable explained by the model.

Based on the output, it appears that Total Debt/Equity and Asset Turnover Ratio are not statistically significant as their p-values are greater than 0.05. The other 11 independent variables, including ROA (return on assets) and the sector variables, appear to be statistically significant in explaining the variation in the dependent variable.

The significantly low R-Squared suggests that most variability in the data was explained by the risk exposures.

#### 2001- Regression 3

Regression using only financial ratios and risk exposure

In [121]:
# General regression code:
X1 = merged_df_2008[['Total Debt/Equity', 'ROA',  'Asset Turnover Ratio']]
X2 = merged_df_2008[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df_2008['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.688
Model:                            OLS   Adj. R-squared:                  0.687
Method:                 Least Squares   F-statistic:                     893.5
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:42   Log-Likelihood:                 3404.0
No. Observations:                2440   AIC:                            -6794.
Df Residuals:                    2433   BIC:                            -6753.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                    0.0208 

This is the output of an Ordinary Least Squares (OLS) regression model with a dependent variable 'RET' (which stands for Return) and six independent variables: Total Debt/Equity, ROA, Asset Turnover Ratio, mktrf, smb, and hml. The model has been estimated using 2,440 observations. The model has an R-squared value of 0.688, indicating that the independent variables explain 68.8% of the variation in the dependent variable, with an adjusted R-Squared of 68.7%, indicating that there is no overfitting. All independent variables except Total Debt/Equity are statistically significant at a 5% level of significance. 

#### 2008- Regression 4

Regression using only risk exposure

In [122]:
#General regression code:
X = merged_df_2008[['mktrf', 'smb', 'hml']]
y = merged_df_2008['RET']
# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

#list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.683
Model:                            OLS   Adj. R-squared:                  0.683
Method:                 Least Squares   F-statistic:                     1749.
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:42   Log-Likelihood:                 3385.0
No. Observations:                2440   AIC:                            -6762.
Df Residuals:                    2436   BIC:                            -6739.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0239      0.003      9.171      0.0

In order to further understand the impact of risk exposures, the below above model is analyzed.

In this multiple linear regression model, the dependent variable is RET (stock returns) with six independent variables: Total Debt/Equity, ROA (return on assets), Asset Turnover Ratio, mktrf (systematic risk), smb (small minus big firms), and hml (high minus low book-to-market ratio firms).

Ths previous model showed that all six independent variables were statistically significant in predicting the dependent variable. The R-squared value was 0.688, which means that 68.8% of the variation in the dependent variable was explained by the independent variables.

This model shows a simplified version of the previous model, with only three independent variables: mktrf, smb, and hml. These three variables are also statistically significant in predicting the dependent variable. The R-squared value is 0.683, which means that 68.3% of the variation in the dependent variable is explained solely by the risk exposures.

#### 2008- Regression 5

Regression with industry sectors

In [142]:
X_all_sectors = pd.get_dummies(merged_df_2008['gsector'], prefix='sector')
y = merged_df_2008['RET']
# Perform the regression using statsmodels
model_all_sectors = sm.OLS(y, X_all_sectors).fit()

# Identify the industry with the coefficient value closest to zero
reference_sector = model_all_sectors.params.abs().idxmin()

print(reference_sector)

sector_55.0


In [143]:
# Create dummy variables for the industry sectors
X_all_sectors = pd.get_dummies(merged_df_2008['gsector'], prefix='sector')
y = merged_df_2008['RET']

# Exclude the reference sector's dummy variable from the independent variables
X = X_all_sectors.drop(columns=[reference_sector])

# Add a constant to the independent variables
X = sm.add_constant(X)

# Perform the regression using statsmodels without the reference sector's dummy variable
model_5 = sm.OLS(y, X).fit()

# Print the regression output
print(model_5.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.110
Model:                            OLS   Adj. R-squared:                  0.107
Method:                 Least Squares   F-statistic:                     30.15
Date:                Sat, 06 May 2023   Prob (F-statistic):           2.70e-55
Time:                        12:37:00   Log-Likelihood:                 2126.4
No. Observations:                2440   AIC:                            -4231.
Df Residuals:                    2429   BIC:                            -4167.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0565      0.011     -4.922      

The regression output shows the results of the OLS regression model that predicts the returns of companies based on their industry sectors. The R-squared value of the model is 0.110, which means that 11% of the variance in the dependent variable (RET) is explained by the independent variables (the dummy variables for industry sectors).

The coefficients for each industry sector are negative, which means that, on average, companies in these sectors have lower returns than the reference sector (which is not included in the model).

The reference sector (secotr 55) is not explicitly mentioned in the output, but it can be inferred from the fact that one dummy variable is excluded from the model. The reference sector is the industry sector that has the coefficient value closest to zero.

The coefficients of the different sectors indicate how much the return of a company is affected by the sector it belongs to, compared to the reference sector. For example, companies in sector 10.0, which is the energy sector, have returns that are 0.1251 lower, on average, than the reference sector, which is Utilities. Similarly, companies in Consumer Discretionary sector (25.0) have returns that are 0.1067 lower, on average, than the reference sector.

The intercept coefficient (const) is -0.0565, which means that, on average, a company in the reference sector has a return of -0.0565 when all other independent variables are held constant.

Overall, the results suggest that industry sector is a significant predictor of company returns, with different sectors having varying levels of impact on returns.

### 2001- Conclusion

**`For the Great Recession of 2008, the most explanatory power is held by the risk exposures.`**

# The Covid Shock 

The COVID-19 pandemic caused a global stock market crash in 2020, with major indices experiencing significant declines in March. Lockdowns and travel restrictions led to supply chain disruptions and reduced consumer spending, negatively impacting many industries. 

### Merging the three files together 

In [123]:
# Renaming tic to TICKER in the gsector file
gsector_2020 = gsector_2020.rename(columns={'tic': 'TICKER'})

In [124]:
# Merge the DataFrames based on the 'TICKER' column
merged_df_2020 = pd.merge(gsector_2020, ratios_2020, on='TICKER')
merged_df_2020 = pd.merge(merged_df_2020, ff_2020, on='TICKER')

In [125]:
merged_df_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2961 entries, 0 to 2960
Data columns (total 38 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gvkey_x               2961 non-null   int64  
 1   fyear_x               2961 non-null   int64  
 2   TICKER                2961 non-null   object 
 3   gsector               2960 non-null   float64
 4   gvkey_y               2961 non-null   int64  
 5   datadate              2961 non-null   object 
 6   fyear_y               2961 non-null   int64  
 7   indfmt                2961 non-null   object 
 8   consol                2961 non-null   object 
 9   popsrc                2961 non-null   object 
 10  datafmt               2961 non-null   object 
 11  curcd                 2961 non-null   object 
 12  act                   2961 non-null   float64
 13  at                    2961 non-null   float64
 14  che                   2961 non-null   float64
 15  cogs                 

## Run OLS regression with the following three main categories 


1.   **Risk Exposures:** This included systematic risk (market risk), SML (Small Minus Big), which is the size risk, and HML (High minus Low), which is the Value vs Growth risk. 

2.   **Financial Ratios:** This includes various financial ratios, including but not limited to Liquidity, Leverage, Asset Turnover Ratio, Net and Gross Profit Margin, ROE and ROA. 

3.   **Industry Indicators:** This indicates which industries had the highest amount of affect on stock returns. 



### 2020- Regression 1

Regression with financial ratios, industry indicators and risk exposure

In [126]:
# General regression code:
X1 = merged_df_2020[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = pd.get_dummies(merged_df_2020['gsector'], prefix='sector')
X3 = merged_df_2020[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2, X3], axis=1)

y = merged_df_2020['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.556
Model:                            OLS   Adj. R-squared:                  0.554
Method:                 Least Squares   F-statistic:                     217.2
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:42   Log-Likelihood:                 2648.2
No. Observations:                2961   AIC:                            -5260.
Df Residuals:                    2943   BIC:                            -5153.
Df Model:                          17                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                  -0.1412    

The multiple linear regression model has an adjusted R-squared value of 0.554, indicating that the model explains around 55% of the variance in the dependent variable. Among the independent variables, only mktrf, smb, and hml have p-values less than 0.05, indicating that they are statistically significant predictors of the dependent variable at a 95% confidence level. The coefficients of these variables are negative, indicating that they have a negative effect on the stock returns. The other independent variables, including const, Gross Profit Margin, Liquidity, at_log, and sector variables, are not statistically significant at a 95% confidence level.

Here we also see that the as risk increased, including market risk, size risk and distress risk, this negatively impacted the stock returns.

However, the model also has some potential issues with multicollinearity and numerical problems.

### 2020- Regression 2

Regression with financial ratios and industry indicators

In [127]:
# General regression code:
X1 = merged_df_2020[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = pd.get_dummies(merged_df_2020['gsector'], prefix='sector')

X = pd.concat([X1, X2], axis=1)

y = merged_df_2020['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.124
Model:                            OLS   Adj. R-squared:                  0.120
Method:                 Least Squares   F-statistic:                     29.83
Date:                Sat, 06 May 2023   Prob (F-statistic):           8.58e-75
Time:                        12:27:42   Log-Likelihood:                 1640.9
No. Observations:                2961   AIC:                            -3252.
Df Residuals:                    2946   BIC:                            -3162.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                  -0.3420    

This is an OLS regression analysis of the dependent variable RET (stock returns) and 14 independent variables. The adjusted R-squared value is 0.120, which means that the model explains 12% of the variance in the dependent variable after adjusting for the number of independent variables. However, only the independent variables Liquidity and sector_30.0 are significant at a 95% confidence level (P < 0.05), whereas the other independent variables are not significant. There might be some numerical problems or strong multicollinearity in the data, indicated by the large condition number of 1.55e+03.

This significanly low explanatory power is a result of removing market, size and distress risk from the regression.

### 2020- Regression 3

Regression with financial ratios and risk exposure

In [128]:
# General regression code:
X1 = merged_df_2020[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = merged_df_2020[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df_2020['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.542
Model:                            OLS   Adj. R-squared:                  0.541
Method:                 Least Squares   F-statistic:                     583.2
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:43   Log-Likelihood:                 2601.5
No. Observations:                2961   AIC:                            -5189.
Df Residuals:                    2954   BIC:                            -5147.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                  -0.0036    

This is a linear regression model with a dependent variable RET (stock returns) and six independent variables: Gross Profit Margin, Liquidity, at_log, mktrf, smb, and hml. The model has an R-squared of 0.542, indicating that 54.2% of the variability in the dependent variable is explained by the independent variables. Only the variables mktrf, smb, and hml have statistically significant coefficients at the 5% level. The constant term is not statistically significant.

As risk exposures were put back into the regression, the model's explanatory power and improved significantly.





### 2020- Regression 4

Regression with only risk exposure

In [129]:
#General regression code:
X = merged_df_2020[['mktrf', 'smb', 'hml']]
y = merged_df_2020['RET']
# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

#list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.542
Model:                            OLS   Adj. R-squared:                  0.541
Method:                 Least Squares   F-statistic:                     1165.
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        12:27:43   Log-Likelihood:                 2599.8
No. Observations:                2961   AIC:                            -5192.
Df Residuals:                    2957   BIC:                            -5168.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0012      0.004      0.318      0.7

The OLS regression model has an R-squared of 0.542, indicating that the model explains 54.2% of the variability in the dependent variable. The coefficients for Gross Profit Margin and Liquidity are not statistically significant at a 5% level. The final model includes the market factor (mktrf), size factor (smb), and value factor (hml), which are all statistically significant at a 5% level. The adjusted R-squared is 0.541, indicating that the model has a good fit. The model is also free from multicollinearity issues as evidenced by the low condition number.

It is also important to note the inverse relationship between risk exposures and stock returns.

### 2020- Regression 5

Regression with only industry indicators

In [151]:
X_all_sectors = pd.get_dummies(merged_df_2020['gsector'], prefix='sector')
y = merged_df_2020['RET']
# Perform the regression using statsmodels
model_all_sectors = sm.OLS(y, X_all_sectors).fit()

# Identify the industry with the coefficient value closest to zero
reference_sector = model_all_sectors.params.abs().idxmin()

print(reference_sector)

sector_35.0


In [153]:
# Create dummy variables for the industry sectors
X_all_sectors = pd.get_dummies(merged_df_2020['gsector'], prefix='sector')
y = merged_df_2020['RET']

# Exclude the reference sector's dummy variable from the independent variables
X = X_all_sectors.drop(columns=[reference_sector])

# Add a constant to the independent variables
X = sm.add_constant(X)

# Perform the regression using statsmodels without the reference sector's dummy variable
model_5 = sm.OLS(y, X).fit()

# Print the regression output
print(model_5.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.116
Model:                            OLS   Adj. R-squared:                  0.113
Method:                 Least Squares   F-statistic:                     38.64
Date:                Sat, 06 May 2023   Prob (F-statistic):           5.20e-72
Time:                        12:50:31   Log-Likelihood:                 1626.8
No. Observations:                2961   AIC:                            -3232.
Df Residuals:                    2950   BIC:                            -3166.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0807      0.007    -11.584      

This is a linear regression model with the dependent variable RET (return) and ten independent variables, which are dummy variables for industry sectors (excluding the reference sector with the coefficient closest to zero). The model has an R-squared value of 0.116, which means that 11.6% of the variation in the dependent variable can be explained by the independent variables.

Here, a one-unit increase in sector_10.0 (Energy Sector) is associated with a 0.1890 decrease in RET (stock return), holding all other independent variables constant.

The reference sector is sector_30.0, which is the Consumer Staples Sector, and its coefficient is not statistically significant with a p-value of 0.945, indicating that there is no significant difference in return between sector_30.0 and the reference sector.

The coefficients of the other independent variables are all statistically significant with p-values less than 0.05.

### 2020- Conclusion

**`As in the models before for the previous 3 stock market crashes, risk exposures explain the most variability in returns for the months leading up to the Covid Crash in 2020.`**

# The end of Easy Money

The Easy Money stock market crash of 2022 occurred due to the Federal Reserve's decision to raise interest rates, leading to a significant drop in stock prices. The crash resulted in significant losses for investors and sparked concerns about the stability of the financial system.

### Merging the three files together 

In [130]:
# Renaming tic to TICKER in the gsector file
gsector_2022 = gsector_2022.rename(columns={'tic': 'TICKER'})

In [131]:
# Merge the DataFrames based on the 'TICKER' column
merged_df_2022 = pd.merge(gsector_2022, ratios_2022, on='TICKER')
merged_df_2022 = pd.merge(merged_df_2022, ff_2022, on='TICKER')

In [132]:
merged_df_2022.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3026 entries, 0 to 3025
Data columns (total 38 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gvkey_x               3026 non-null   int64  
 1   fyear_x               3026 non-null   int64  
 2   TICKER                3026 non-null   object 
 3   gsector               3025 non-null   float64
 4   gvkey_y               3026 non-null   int64  
 5   datadate              3026 non-null   object 
 6   fyear_y               3026 non-null   int64  
 7   indfmt                3026 non-null   object 
 8   consol                3026 non-null   object 
 9   popsrc                3026 non-null   object 
 10  datafmt               3026 non-null   object 
 11  curcd                 3026 non-null   object 
 12  act                   3026 non-null   float64
 13  at                    3026 non-null   float64
 14  che                   3026 non-null   float64
 15  cogs                 

## Run OLS regression with the following three main categories 


1.   **Risk Exposures:** This included systematic risk (market risk), SML (Small Minus Big), which is the size risk, and HML (High minus Low), which is the Value vs Growth risk. 

2.   **Financial Ratios:** This includes various financial ratios, including but not limited to Liquidity, Leverage, Asset Turnover Ratio, Net and Gross Profit Margin, ROE and ROA. 

3.   **Industry Indicators:** This indicates which industries had the highest amount of affect on stock returns. 



#### 2022- Regression 1 

Regression with financial ratios, industry indicators and risk exposures

In [133]:
# General regression code:
X1 = merged_df_2022[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = pd.get_dummies(merged_df_2022['gsector'], prefix='sector')
X3 = merged_df_2022[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2, X3], axis=1)

y = merged_df_2022['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.231
Model:                            OLS   Adj. R-squared:                  0.227
Method:                 Least Squares   F-statistic:                     53.20
Date:                Sat, 06 May 2023   Prob (F-statistic):          1.67e-157
Time:                        12:27:43   Log-Likelihood:                 3866.7
No. Observations:                3026   AIC:                            -7697.
Df Residuals:                    3008   BIC:                            -7589.
Df Model:                          17                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                   0.0311    

The OLS regression model shows that the R-squared value is 0.231, indicating that the model explains only 23.1% of the variation in the dependent variable, with an adjusted R-Squared of 22.7%.

The regression coefficients for Gross Profit Margin, sector_10.0, sector_15.0, sector_20.0, sector_25.0, sector_30.0, sector_35.0, sector_40.0, sector_45.0, sector_50.0, sector_55.0, and sector_60.0 are not statistically significant at the 5% level. The regression coefficients for Liquidity, at_log, mktrf, smb, and hml are statistically significant at the 5% level. 

The model's goodness of fit can be improved, as indicated by the Adjusted R-squared value of 0.227. The coefficients of smb and hml are negative and positive, respectively, suggesting that a higher SMB (size factor) and HML (value factor) are associated with lower and higher stock market returns, respectively. 

The model has some potential issues with multicollinearity, as indicated by the large condition number.

#### 2022- Regression 2

Regression with only financial ratios and industry indicators.

In [134]:
# General regression code:
X1 = merged_df_2022[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = pd.get_dummies(merged_df_2022['gsector'], prefix='sector')

X = pd.concat([X1, X2], axis=1)

y = merged_df_2022['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.058
Model:                            OLS   Adj. R-squared:                  0.054
Method:                 Least Squares   F-statistic:                     13.34
Date:                Sat, 06 May 2023   Prob (F-statistic):           3.06e-31
Time:                        12:27:43   Log-Likelihood:                 3560.0
No. Observations:                3026   AIC:                            -7090.
Df Residuals:                    3011   BIC:                            -7000.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                   0.0067    

The R-squared value of the model is 0.058, indicating that only 5.8% of the variation in the dependent variable is explained by the independent variables. The Adjusted R-squared value is slightly lower at 0.054, suggesting that the model's goodness of fit could be improved.

The coefficients of the independent variables are mostly negative, indicating that all sectors and risk exposures, except for sml, lead to lower stock market returns. However, the p-values of most of these variables are greater than 0.05, suggesting that they are not statistically significant predictors of stock market returns.

The p-value of at_log is less than 0.05, indicating that it is a statistically significant predictor of stock market returns. This suggests that as a company's assets grow, its stock market returns also increase.

The p-values of the sector indicators are mixed, with some being statistically significant predictors of stock market returns (e.g., sector_15.0, which is Materials) and others not (e.g., sector_10.0, which is Energy).

The model's condition number is large, indicating the possible presence of strong multicollinearity or other numerical problems.

#### 2022- Regression 3

Regression with financial ratios and risk exposures

In [135]:
# General regression code:
X1 = merged_df_2022[['Gross Profit Margin', 'Liquidity', 'at_log']]
X2 = merged_df_2022[['mktrf', 'smb', 'hml']]

X = pd.concat([X1, X2], axis=1)

y = merged_df_2022['RET']

# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

# list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.211
Model:                            OLS   Adj. R-squared:                  0.209
Method:                 Least Squares   F-statistic:                     134.2
Date:                Sat, 06 May 2023   Prob (F-statistic):          4.77e-151
Time:                        12:27:43   Log-Likelihood:                 3826.7
No. Observations:                3026   AIC:                            -7639.
Df Residuals:                    3019   BIC:                            -7597.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                  -0.0188    

The results indicate that the model has an R-squared value of 0.211, which implies that approximately 21.1% of the variance in the dependent variable can be explained by the independent variables. The F-statistic of 134.2 is statistically significant with a p-value of 4.77e-151. The results also indicate that some of the independent variables are statistically significant, while others are not.

Among the statistically significant variables, mktrf, smb, and hml are significant at a 95% level of significance. This implies that these variables have a strong impact on the dependent variable, RET. Specifically, mktrf has a negative coefficient of -0.0495, which means that an increase in the market risk premium will lead to a decrease in the firm's return on equity. 

Similarly, smb has a negative coefficient of -0.0135, while hml has a positive coefficient of 0.0322, indicating that an increase in the size and value factors will lead to a decrease and increase, respectively, in the firm's returns.



#### 2022- Regression 4

Regression with only the risk exposures

In [136]:
#General regression code:
X = merged_df_2022[['mktrf', 'smb', 'hml']]
y = merged_df_2022['RET']
# Use statsmodels
X = sm.add_constant(X) # adding a constant
model = sm.OLS(y, X).fit()

#list regression output
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.207
Model:                            OLS   Adj. R-squared:                  0.206
Method:                 Least Squares   F-statistic:                     262.3
Date:                Sat, 06 May 2023   Prob (F-statistic):          2.75e-151
Time:                        12:27:44   Log-Likelihood:                 3819.1
No. Observations:                3026   AIC:                            -7630.
Df Residuals:                    3022   BIC:                            -7606.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.0070      0.003     -2.625      0.0

The R-squared value of the model is 0.207, which indicates that the independent variables explain approximately 21% of the variation in the dependent variable. The adjusted R-squared value is 0.206, which accounts for the degrees of freedom in the model.

The coefficients of the independent variables show the effect of each variable on the dependent variable, holding other variables constant. The constant is -0.0070, indicating that the expected value of the dependent variable when all independent variables are equal to zero is -0.0070.

The coefficients of "mktrf", "smb", and "hml" are -0.0491, -0.0130, and 0.0308, respectively. These coefficients suggest that "mktrf" and "smb" have a negative effect on the dependent variable, while "hml" has a positive effect. All three coefficients are statistically significant with p-values less than 0.05.

#### 2022- Regression 5

Regression with only the industry indicators 

In [146]:
X_all_sectors = pd.get_dummies(merged_df_2022['gsector'], prefix='sector')
y = merged_df_2022['RET']
# Perform the regression using statsmodels
model_all_sectors = sm.OLS(y, X_all_sectors).fit()

# Identify the industry with the coefficient value closest to zero
reference_sector = model_all_sectors.params.abs().idxmin()

print(reference_sector)

sector_55.0


In [147]:
# Create dummy variables for the industry sectors
X_all_sectors = pd.get_dummies(merged_df_2022['gsector'], prefix='sector')
y = merged_df_2022['RET']

# Exclude the reference sector's dummy variable from the independent variables
X = X_all_sectors.drop(columns=[reference_sector])

# Add a constant to the independent variables
X = sm.add_constant(X)

# Perform the regression using statsmodels without the reference sector's dummy variable
model_5 = sm.OLS(y, X).fit()

# Print the regression output
print(model_5.summary())

                            OLS Regression Results                            
Dep. Variable:                    RET   R-squared:                       0.053
Model:                            OLS   Adj. R-squared:                  0.049
Method:                 Least Squares   F-statistic:                     16.71
Date:                Sat, 06 May 2023   Prob (F-statistic):           8.51e-30
Time:                        12:49:18   Log-Likelihood:                 3550.5
No. Observations:                3026   AIC:                            -7079.
Df Residuals:                    3015   BIC:                            -7013.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0157      0.008     -1.908      

This is a regression analysis for the year 2022 using the Ordinary Least Squares (OLS) method. The dependent variable is the 'RET' column, and the independent variables are the dummy variables for the 'gsector' column of the 'merged_df_2022' dataset.

The regression output shows that the R-squared value is 0.053, which means that the model explains only 5.3% of the variance in the dependent variable.

The coefficient values for the different industry sectors indicate the impact of each sector on the dependent variable. The 'sector_15.0' industry has the most significant impact with a coefficient value of -0.0599, which means that a one-unit increase in this sector's dummy variable decreases the dependent variable by 0.0599 units, holding all other variables constant. The reference sector, 'sector_10.0', has a coefficient value of -0.0095, which is not statistically significant.

The model's overall fit can be improved as the adjusted R-squared value is only 0.049.

Overall, this regression analysis suggests that the industry sector has a statistically significant impact on the dependent variable, but the model's fit is not strong enough to make accurate predictions.

### 2022- Conclusion

**`For the End of Easy Money Stock Market Crash, the most variability in the stock returns are explained by the risk exposures`**

# Links and References 

1. [https://en.wikipedia.org/wiki/
Global_Industry_Classification_Standard](https://)
2. https://www.investopedia.com/timeline-of-stock-market-crashes-5217820
3. https://en.wikipedia.org/wiki/List_of_stock_market_crashes_and_bear_markets
4. https://www.bankrate.com/investing/biggest-stock-market-crashes-in-us-history/